src/inmanta/plugins.py

```diff
@@ -277,26 +281,32 @@ def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
 
         # dict
         if issubclass(origin, collections.abc.Mapping):
-            args = typing_inspect.get_args(python_type)
-            if not args:
-                return inmanta_type.TypedDict(inmanta_type.Type())
+            if origin in [collections.abc.Mapping, dict, typing.Mapping]:
+                args = typing_inspect.get_args(python_type)
+                if not args:
+                    return inmanta_type.TypedDict(inmanta_type.Type())
 
-            if not issubclass(args[0], str):
-                raise TypingException(
-                    None, f"invalid type {python_type}, the keys of any dict should be 'str', got {args[0]} instead"
-                )
+                if not issubclass(args[0], str):
+                    raise TypingException(
+                        None, f"invalid type {python_type}, the keys of any dict should be 'str', got {args[0]} instead"
+                    )
 
-            if len(args) == 1:
-                return inmanta_type.TypedDict(inmanta_type.Type())
+                if len(args) == 1:
+                    return inmanta_type.TypedDict(inmanta_type.Type())
 
-            return inmanta_type.TypedDict(to_dsl_type(args[1]))
+                return inmanta_type.TypedDict(to_dsl_type(args[1]))
+            else:
+                raise TypingException(None, f"invalid type {python_type}, dictionary types should be Mapping or dict")
 
         # List
         if issubclass(origin, collections.abc.Sequence):
-            args = typing.get_args(python_type)
-            if not args:
-                return inmanta_type.List()
-            return inmanta_type.TypedList(to_dsl_type(args[0]))
+            if origin in [collections.abc.Sequence, list, typing.Sequence]:
```

Comment:
Not Collection?

Reply:
Collection also contains Mapping and Set, so I think that is a bit too wide?

Reply:
But we're not accepting subclasses anymore, only those exact annotations. A list is a collection, so that would be acceptable for input args. And we can easily construct lists from it so it's acceptable as return type.

But I'm not too attached to it, it's not that common.